protopipe/pipeline/data_writer.py

```diff
@@ -0,0 +1,591 @@
+#!/usr/bin/env python3
+"""
+Class to write data from an event stream
+"""
+
+
+import pathlib
+from collections import defaultdict
+from typing import DefaultDict, Tuple
+
+import numpy as np
+import tables
+from astropy import units as u
+
+from ..containers import (
+    ArrayEventContainer,
+    SimulatedShowerDistribution,
+    TelEventIndexContainer,
+)
+from ..core import Component, Container, Field, Provenance, ToolConfigurationError
+from ..core.traits import Bool, CaselessStrEnum, Int, Path, Float, Unicode
+from ..io import EventSource, HDF5TableWriter, TableWriter
+from ..io.simteleventsource import SimTelEventSource
+from ..io import metadata as meta
+from ..io.tableio import FixedPointColumnTransform
+from ..instrument import SubarrayDescription
+
+__all__ = ["DataWriter", "DL1_DATA_MODEL_VERSION", "write_reference_metadata_headers"]
+
+tables.parameters.NODE_CACHE_SLOTS = 3000  # fixes problem with too many datasets
+
+DL1_DATA_MODEL_VERSION = "v1.1.0"
```

Comment:
Should remove "DL1" from all of these. This will define a complete data model (e.g. DATA_MODEL_VERSION)

Reply:
"resolving", since this code will disappear soon with the next commits